client/app/pages/dashboards/DashboardPage.jsx

```diff
@@ -412,10 +412,10 @@ DashboardPage.defaultProps = {
 
 export default {
   path: "/dashboard/:dashboardSlug",
-  render: routeParams => (
-    <AuthenticatedPageWrapper key={`/dashboard/${routeParams.dashboardSlug}`}>
+  render: currentRoute => (
+    <AuthenticatedPageWrapper key={currentRoute.key}>
       <ErrorBoundaryContext.Consumer>
-        {({ handleError }) => <DashboardPage {...routeParams} onError={handleError} />}
+        {({ handleError }) => <DashboardPage {...currentRoute.routeParams} onError={handleError} />}
       </ErrorBoundaryContext.Consumer>
     </AuthenticatedPageWrapper>
   ),
```

Comment:
Not sure if it's that worth it at this point, but it looks like pages follow the same pattern when being defined, if this is not intended to be changed (and every page should follow this notation), we perhaps could abstract it a bit more with a closure returning this function? (seems we can tweak it to receive only the page component and some extra routeParams):

```js
{
  path: "/dashboard/:dashboardSlug",
  render: authenticatedPage(DashboardPage, { extraData: "dashboard-page" }),
}
```

BTW, if I had to choose, I think it's better to pass `routeParams` as separate props (like you did to List Pages) and not directly to the props (I don't know, but it feels risky to me to leave props that open :grimacing:).

Well, lmk your opinion

Reply:
1. Yes, indeed it makes sense. But let's do it in a follow-up PR - I don't want to add anything more to this one (except of bug fixes related to PR's subject);
2. With new router it's safe to spread route params - that object contains only values from URL (in this example - `dashboardSlug`) and resolved values (which will be removed soon), Router does not add anything to it (unlike Angular's router which was merging params from URL, resolved values and query string params).